draft/0005-impersonation.md

```diff
@@ -0,0 +1,260 @@
+---
+Start date: 2024-05-30
+Author(s): wangsijie
+Revision: 1
+---
+
+# Impersonation
+
+## Overview
+
+User impersonation is a feature that allows an authorized user to assume the identity of another user within the system. This capability can be useful for various reasons such as troubleshooting, providing support, and performing administrative tasks. This document aims to explore the design and implementation of user impersonation in Logto.
+
+## Motivation
+
+Logto is an enterprise ready identity provider, and it is essential to provide user impersonation capabilities to support various use cases. For example, a support engineer may need to troubleshoot an issue on behalf of a user, or an administrator may need to verify the permissions of a user. User impersonation can help to streamline these processes and improve the overall user experience.
+
+## Token exchange
```

Comment:
before this section, it'll be great to have an overview for the impersonation flow (calling management api, token exchange, and use the exchanged token), e.g. some introduction with a [sequence diagram](https://mermaid.js.org/syntax/sequenceDiagram.html)